### ISSUE TITLE
Trusted setup plans & check-list

### ISSUE THREAD

**Issue by cwgoes** [anoma/spec#20]
Per discussion in https://github.com/cryptiumlabs/cryptography/issues/17#issuecomment-587064460 and elsewhere, we want to run a trusted setup this summer, for two systems in parallel:

- Groth16 on BLS12-381 for the multi-asset shielded pool circuit
  This setup just needs to handle the MASP circuit, since it is circuit-specific.
- Kate commitments on BLS12-381 for PLONK/plookup/Marlin, for use with the Juvix arithmetic circuit backend and the BLS12-381 opcodes which are slated for inclusion in Tezos in 007.
  This setup should be quite large in order to handle future circuits.

Code-wise, we need to:
- [ ] Modify the Powers of Tau setup (second part) for use with the updated MASP circuit
- [ ] Check if we can easily modify the Aztec Kate commitment setup for use with BLS12-381
- [ ] Combine the two into an easily-scriptable, user-friendly (user-friendlier) software package

Then we should plan to run this setup late summer (mainly blocked on testing/audits of MASP circuit).

**Comment by cwgoes** [comment]
Closing in favour of more recent tracking issues.
